Learn - Data Transfer  Update (#511)
iota-community/iota-wiki: internal/learn/about-iota/data-transfer.md

Clients, which can be wallets or applications, send and receive messages (data objects) through nodes in IOTA. Nodes are the entry and exit points for these messages, and those nodes communicate with each other and with the connected clients.

> There are several types of messages implemented in IOTA. Some messages transfer value (the IOTA token or digital assets), while others transfer only pure data, and some message types can even contain both values and data. This flexible message structure enables the decentralized transport of data and value in a single message with the highest security grade and the total absence of fees. The network nodes take care of the secure distribution of all those messages in the Tangle.

Reviewer: As a non-English reader, this now reads as if values is meant as data values and not value as economic value.

Author: Ah, this was only to make it more "plural" friendly. I can rework it.

:::note

In the upcoming 2.0 IOTA version, this PoW requirement will become adaptive and is planned to be completely removed for times of normal network loads. You can read more about [IOTA 2.0 here](/build/networks/iota-2.0).

:::

## Message Structure

This is the defined syntactical structure of every message in the IOTA protocol (the label of our package):

<table>
    <tr>
        <th>Name</th>

>         <th>Type</th>x

Reviewer: ```suggestion
        <th>Type</th>
```

I guess that was a typo 😄

Author: 😭

        <th>Description</th>
    </tr>
    <tr>
        <td>Protocol Version</td>
        <td>uint8</td>
        <td>Protocol version number of the message.</td>
        </tr>
    <tr>
        <td> Parents Count </td>
        <td> uint8</td>
        <td> The number of messages that are directly approved.</td>
    </tr>
    <tr>
        <td>Parents <code>anyOf</code> </td>
        <blockquote>
        References another directly approved message.
        </blockquote>
        <td>ByteArray[32] </td>
        <td>The Message ID of the parent.</td>
    </tr>
    <tr>
        <td>Payload Length</td>
        <td>uint32</td>

>         <td> The length of the following payload in bytes. A length of 0 means no payload will be attached.</td>

Reviewer: ```suggestion
        <td>The length of the following payload in bytes. A length of 0 means no payload will be attached.</td>
```

## Message Structure

:::note

Protocol Version and uint8 have replaced the formed NetworkID and uint64 ahead of the Stardust release.

:::

> 

Reviewer: I think there is a bit of misunderstand what I mean with the "single version" of these docs and indicating what changed. I don't mean "describe what has been changed for the Stardust release", I mean we keep the main documentation reflecting the current state of mainnet and then use notes, sections, tabs, etc to document the difference in Shimmer, IOTA 2.0, etc. For example something like (just to get the idea):

```
The NetworkID is a uint64, indicating what network this message is meant for.

:::note
On Shimmer the NetworkID is renamed to Protocol Version and is now a uint8.
:::
```

### Message Validation

:::note

The message validation rules have been updated to reflect the upcoming Stardust release.

:::

> 

Reviewer: Same here. I'd not describe what has been changed. I'd document the actual change and leave the documentation below related to mainnet.

| Payload Name                                                                                                    | Type Value |
| --------------------------------------------------------------------------------------------------------------- | ---------- |

> | [Tagged Date](https://hackmd.io/TIP-0023/tip-0023.md)                                                           | 5          |

Reviewer 2: ```suggestion
| [Tagged Data](https://hackmd.io/TIP-0023/tip-0023.md)                                                           | 5          |
```

| [Transaction](https://hackmd.io/TIP-0020/tip-0020.md)                                                           | 6          |

> | [Milestone](https://hackmd.io/TIP-0029/tip-0029.md)                                                             | 7          |

Reviewer 2: change links to links of the tips repo.

Reviewer 2: PRs are ok

> A message containing only an **indexation payload** (Data) can be sent without a signature. It can hold any data the user wants to send, as long as it is parsable and follows the required syntax and size limit.

Reviewer 2: ```suggestion
A message containing only a **tagged data payload** (Data) can be sent without a signature. It can hold any data the user wants to send, as long as it is parsable and follows the required syntax and size limit.
```

### Payloads

:::note

The three payload types have been updated for the upcoming Stardust release. You can read more about the changes in our [Messages](internal/learn/about-iota/messages.md) section.

:::

> 

Reviewer: Same here. I'd say something like:
```
:::note
The payload types are different on Shimmer. You can read more about the changes in the [Messages](internal/learn/about-iota/messages.md) section.
:::
```
